include/image.mk

```diff
@@ -381,13 +381,13 @@ define Device/Init
   DEVICE_NAME := $(1)
   KERNEL:=
   KERNEL_INITRAMFS = $$(KERNEL)
-  KERNEL_SIZE:=
   CMDLINE:=
 
   IMAGES :=
   ARTIFACTS :=
   IMAGE_PREFIX := $(IMG_PREFIX)-$(1)
   IMAGE_NAME = $$(IMAGE_PREFIX)-$$(1)-$$(2)
+  IMAGE_SIZE :=
```

Comment:
If Device/Init is actually setting per-device global variables, that might be the solution for your problems with DEVICE_TYPE. Just move the global default DEVICE_TYPE here and add it to DEVICE_VARS ...

Reply:
Yes, I tried it in my draft patch and it seems to work fine. Thanks for the tip. :)